apps/devhub/tests/test_views.py

```diff
@@ -2871,6 +2897,30 @@ def test_unlisted_addon_pass_validation(self, mock_sign_file):
         assert file_.status == amo.STATUS_LITE
         assert mock_sign_file.called
 
+    @mock.patch('devhub.views.sign_file')
+    def test_experiments_are_auto_signed(self, mock_sign_file):
+        """Experiment extensions (bug 1220097) are auto-signed."""
+        self.upload = self.get_upload('experiment.xpi')
+        self.addon.update(guid='experiment@xpi', is_listed=True,
+                          status=amo.STATUS_PUBLIC, trusted=False)
+        # We're going to sign even if it has errors/warnings.
+        self.upload.update(
+            validation=json.dumps({
```

Comment:
It looks like you can pass `validation` to `self.get_upload`. It looks to me like even though this is setting the validation to have errors `self.upload.valid` might still be `True` since the upload was created [without validation errors](https://github.com/mozilla/olympia/blob/9a9231e0c79b99e0ac0e42e74d6b65a970c0176a/apps/files/tests/test_models.py#L52) and [valid will only get set to `True`](https://github.com/mozilla/olympia/blob/9a9231e0c79b99e0ac0e42e74d6b65a970c0176a/apps/files/models.py#L605) not `False` when `validation` changes.